src/api/v1/authorization/routes/login.ts

```diff
@@ -0,0 +1,27 @@
+import { FastifyReply, FastifyRequest } from 'fastify'
+import { checkPassword } from '../../../../security/passwords'
+
+export default async (req: FastifyRequest<{ Body: { username: string, password: string } }>, res: FastifyReply): Promise<void> => {
+  if (!req.body.username) {
+    await res.status(400).send({ message: 'User not specified' })
+  } else {
+    const user = await prisma.user.findFirst({ where: { username: req.body.username } })
+
+    if (!user) {
+      await res.status(401).send({ message: 'Invalid username or password' })
+    } else {
+      if (!req.body.password) {
+        await res.status(400).send({ message: 'Password not specified' })
```

Comment:
Might be worth validating this together with username. This now introduces unnecessary code splitting.